core-patch/src/test/scala/eu/ostrzyciel/jelly/core/patch/PatchDecoderSpec.scala

```diff
@@ -0,0 +1,354 @@
+      val e = intercept[RdfProtoDeserializationError] {
+        decoder.ingestFrame(input)
+      }
+      e.getMessage should include("Incoming stream with statement type")
+      e.getMessage should include("cannot be decoded by this decoder")
+    }
+  }
+
+  "QuadsDecoder" should {
+    "decode quads (1)" in {
+      val input = PatchTestCases.Quads1.encodedFull(
+        JellyPatchOptions.smallStrict
+          .withStatementType(PatchStatementType.QUADS)
+          .withStreamType(PatchStreamType.FLAT),
+        10_000
+      ).head
+      val out = PatchCollector()
+      val decoder = MockPatchConverterFactory.quadsDecoder(out, None)
+      decoder.ingestFrame(input)
+      out.statements.result() shouldBe PatchTestCases.Quads1.mrl
+    }
+
+    "not accept a stream with statement type TRIPLES" in {
+      val input = RdfPatchFrame(Seq(
+        RdfPatchRow.ofOptions(JellyPatchOptions.smallStrict
+          .withStatementType(PatchStatementType.TRIPLES)
+          .withStreamType(PatchStreamType.FLAT)
+        ),
+      ))
+      val decoder = MockPatchConverterFactory.quadsDecoder(null, None)
+      val e = intercept[RdfProtoDeserializationError] {
+        decoder.ingestFrame(input)
+      }
+      e.getMessage should include("Incoming stream with statement type")
+      e.getMessage should include("cannot be decoded by this decoder")
+    }
+  }
+
+  "AnyStatementDecoder" should {
+    "return no options if the inner decoder is not initialized" in {
+      val out = PatchCollector()
+      val decoder = MockPatchConverterFactory.anyStatementDecoder(out, None)
+      decoder.getPatchOpt shouldBe None
+    }
+
+    "throw exception if the first row in the stream is not options" in {
+      val input = RdfPatchFrame(Seq(
+        RdfPatchRow.ofStatementAdd(RdfQuad()),
+      ))
+      val out = PatchCollector()
+      val decoder = MockPatchConverterFactory.anyStatementDecoder(out, None)
+      val e = intercept[RdfProtoDeserializationError] {
+        decoder.ingestFrame(input)
+      }
+      e.getMessage should include("The first row in the stream must be an RdfPatchOptions row")
+    }
+
+    "ignore multiple options rows" in {
+      val opt = JellyPatchOptions.smallStrict
+        .withStatementType(PatchStatementType.TRIPLES)
+        .withStreamType(PatchStreamType.FLAT)
+      val input = RdfPatchFrame(Seq(
+        RdfPatchRow.ofOptions(opt),
+        RdfPatchRow.ofOptions(opt),
+        RdfPatchRow.ofOptions(opt),
+      ))
+      val out = PatchCollector()
+      val decoder = MockPatchConverterFactory.anyStatementDecoder(out, None)
+      decoder.ingestFrame(input)
+      decoder.getPatchOpt shouldBe Some(opt)
+    }
+
+    "decode quads" in {
+      val input = PatchTestCases.Quads1.encodedFull(
+        JellyPatchOptions.smallStrict
+          .withStatementType(PatchStatementType.QUADS)
+          .withStreamType(PatchStreamType.FLAT),
+        10_000
```

Comment:
you're using this specific value of 10_000 many times, don't you want to set it as a val earlier in the tests

Reply:
No, it's tests. The value doesn't matter, it just needs to be big.